### PR TITLE
fix: resolve key access error when sources are not present in manifest

### LINKED ISSUE #641: Maybe revise the hard key access for `source` nodes in dbt integration
**Describe the bug**
If a `manifest.json` does not have nodes of type `sources` the parsing can throw a key error:
```
11:40:11.599 INFO  SodaSqlCon | STDERR:   File "/usr/local/lib/python3.8/site-packages/sodasql/dbt.py", line 83, in parse_manifest
11:40:11.600 INFO  SodaSqlCon | STDERR:     for source_name, source in manifest["sources"].items()
11:40:11.600 INFO  SodaSqlCon | STDERR: KeyError: 'sources'
```

We might want to do a "soft" access via a `.get()` if possible.

I didn't do further research into how feasible this is but if no one picks this up I'll have a closer look as soon as I have a minute

The issue was highlighted by @Antoninj who was writing some integration tests and had an outdated test file which did not contain any `source` nodes.

### PR DESCRIPTION
**Pull request by bastienboutonnet**
Resolves #641

### CODE REVIEW

File: core/sodasql/cli/ingest.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class Table:
 
 
 def map_dbt_run_result_to_test_result(
-    test_nodes: dict[str, "DbtTestNode"],
+    test_nodes: Optional[dict[str, "DbtTestNode"]],
```
Comment (vijaykiran) [inline core/sodasql/cli/ingest.py:38]: Looks like `Optional` isn't imported, can you add it?

Alternatively, you can also use shortcut syntax here `dict[str, "DbtTestNode"] | None`

File: packages/dbt/sodasql/dbt.py [hunk 2]
```diff
@@ -25,15 +25,15 @@
 def parse_manifest(
     manifest: dict[str, Any]
 ) -> tuple[
-    dict[str, ParsedModelNode | CompiledModelNode],
-    dict[str, ParsedSeedNode | CompiledSeedNode],
-    dict[str, ParsedGenericTestNode | CompiledGenericTestNode],
-    dict[str, ParsedSourceDefinition],
+        Optional[dict[str, ParsedModelNode | CompiledModelNode]],
```
Comment (vijaykiran) [inline packages/dbt/sodasql/dbt.py:28]: Same comment as above for `Optional`s here

File: packages/dbt/sodasql/dbt.py [hunk 3]
```diff
@@ -114,10 +118,9 @@ def parse_run_results(run_results: dict[str, Any]) -> list[RunResultOutput]:
     parsed_run_results = [RunResultOutput(**result) for result in run_results["results"]]
     return parsed_run_results
 
-
 def create_nodes_to_tests_mapping(
     model_nodes: dict[str, ParsedModelNode],
-    test_nodes: dict[str, CompiledGenericTestNode],
+    test_nodes: Optional[dict[str, CompiledGenericTestNode | ParsedGenericTestNode]],
```
Comment (vijaykiran) [inline packages/dbt/sodasql/dbt.py:123]: Same as above :)

**Review by vijaykiran — CHANGES_REQUESTED** [review]

File: core/sodasql/cli/ingest.py (hunk 1 above)
Comment (bastienboutonnet) [reply core/sodasql/cli/ingest.py:38]: That's strange that I didn't get an error here...
I'll change it to the more modern (i.e. without `Optional` though as I think it's better).

**Pull request merged**